build_all.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -e -u -o -x pipefail
+
+KUBEKITE_IMAGE_BASE=us.gcr.io/sigma-1330/kubekite
+TEMPLATES=job-templates/*
+REGEX='.*\/(.*)\..*'
+
+# set +u
+
+for TEMPLATE_FILENAME in $TEMPLATES
+do
+    if [[ $TEMPLATE_FILENAME =~ $REGEX ]]
+    then
+        TEMPLATE_NAME=${BASH_REMATCH[1]}
```

Comment:
What does `BASH_REMATCH` do here?

Reply:
`$TEMPLATE_FILENAME =~ $REGEX` is bash's regex syntax, it saves to `BASH_REMATCH`. I wanted to use `grep` or `expr` originally, but it seems like OSX doesn't have versions of them with regex.